### PR TITLE
initial version with resample

### PR DESCRIPTION
**Pull request by stsci-hack**
This enables resampling in CALIMAGE2 and now requires a VALID WCS in order for the resampling to work.

### CODE REVIEW

File: jwst/pipeline/calwebb_image2.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from ..assign_wcs import assign_wcs_step
 from ..flatfield import flat_field_step
 from ..photom import photom_step
+from ..resample import resample_step
 
 
 __version__ = "3.0"
```
Comment (hbushouse) [inline jwst/pipeline/calwebb_image2.py]: This version number is probably in conflict with what's in the repo now (0.8.0).

File: jwst/pipeline/calwebb_image2.py [hunk 2]
```diff
@@ -126,6 +128,12 @@ def process_exposure_product(
         input = self.flat_field(input)
         input = self.photom(input)
 
+        # Resample individual exposures
+        result = self.resample(input)
+
+        # write out resampled exposure
+        self.save_model(result, suffix='i2d')
```
Comment (hbushouse) [inline jwst/pipeline/calwebb_image2.py]: Might need to put this in an IF-block and only call it if `result` exists. In the event a user skips the resample step, it won't exist.

**Comment by hbushouse** [comment]
Fixes #687.

File: jwst/pipeline/calwebb_image2.py (hunk 2 above)
Comment (stsci-hack) [reply jwst/pipeline/calwebb_image2.py]: Latest commit addresses this last comment.

**Comment by jdavies-st** [comment]
Simulated NIRCam data ran through the `calwebb_image2` pipeline without problem, producing a nice rectified `i2d` resampled output.  I did notice that that flux values are much too high, but I think this is related to issue #1381, which can be tracked there.  This is OK to merge.

**Pull request merged**